proxy_http.go

```diff
@@ -111,11 +134,14 @@ func (proxy *proxy) processRequests(ctx filters.Context, remoteAddr string, req
 		upstreamAddr := upstreamAddr(ctx)
 		isConnect := upstream != nil || upstreamAddr != ""
 
+		buffered := downstreamBuffered.Buffered()
```

Comment:
Is it to fix an existing bug? I guess it's useful only when `isConnect` is true?

Reply:
It's been a while since I worked on this change now, but IIRC with the addition of MITM, I ran into a scenario where the client send data following the CONNECT request which we end up having read into the buffer already. In practice, I don't think I ever saw this happen without MITM, but in a sense it is a potential existing bug.